pkg/api/dataproxy.go

```diff
@@ -18,7 +19,34 @@ import (
 	"github.com/grafana/grafana/pkg/util"
 )
 
+type cachedTransport struct {
+	transport *http.Transport
+	updated   time.Time
+}
+
+type proxyTransportCache struct {
+	cache map[int64]*cachedTransport
+	sync.Mutex
+}
+
+var ptc = proxyTransportCache{
+	cache: make(map[int64]*cachedTransport),
+}
+
 func DataProxyTransport(ds *m.DataSource) (*http.Transport, error) {
+	ptc.Lock()
+	defer ptc.Unlock()
+
+	var tlsAuth, tlsAuthWithCACert bool
+	if ds.JsonData != nil {
+		tlsAuth = ds.JsonData.Get("tlsAuth").MustBool(false)
+		tlsAuthWithCACert = ds.JsonData.Get("tlsAuthWithCACert").MustBool(false)
+	}
+
+	if t, present := ptc.cache[ds.Id]; present && t.transport.TLSClientConfig.InsecureSkipVerify != tlsAuth && t.updated == ds.Updated {
```

Comment:
*it's probably easier to make a new transport instead of worrying about getting the boolean comparison right